content/stellar-contracts/tokens/rwa/rwa.mdx

```diff
@@ -0,0 +1,512 @@
+## Modules
+
+The RWA package includes several supporting modules that work together to provide comprehensive regulatory compliance:
+
+### - Compliance
+[Source Code](https://github.com/OpenZeppelin/stellar-contracts/tree/main/packages/tokens/src/rwa/compliance)
+
+This is a mandatory module, since RWA token contract expects the compliance checks and hooks to be available.
+
+Provides a modular framework for implementing custom compliance rules through a hook-based architecture where multiple
+[compliance modules](https://github.com/OpenZeppelin/stellar-contracts/blob/main/packages/tokens/src/rwa/compliance/mod.rs#L327)
+can be registered to the [Compliance Contract](https://github.com/OpenZeppelin/stellar-contracts/blob/main/packages/tokens/src/rwa/compliance/mod.rs#L67).
+
+**Compliance Validation Flow:**
+
+```mermaid
+graph TD
+    A[RWA Token Transfer/Mint] --> B[Compliance Contract]
+    B --> C{CanTransfer<br/>or<br/>CanCreate}
+    C --> D[Registered Modules 1..N]
+    D --> E[Transfer Limit Module]
+    D --> F[Country Restriction Module]
+    D --> G[Investor Count Module]
+    D --> H[Custom Module X]
+    E --> I{All modules<br/>return true?}
+    F --> I
+    G --> I
+    H --> I
+    I -->|Yes| J[Operation Proceeds]
+    I -->|No| K[Operation Reverts]
+```
+
+The compliance contract is designed to be shared across multiple RWA tokens, with each hook function accepting a
+`token` parameter to identify the calling token.
+
+### - Identity Verifier
+[Source Code](https://github.com/OpenZeppelin/stellar-contracts/tree/main/packages/tokens/src/rwa/identity_verifier)
+
+This is a mandatory module, since RWA token contract expects `verify_identity(e: &Env, address: Address)`
+function to be available.
+
+The **Identity Verifier Module** provides the interface for verifying user identities.
+It can also support possible custom implementation approaches:
+
+- **Claim-based**: Cryptographic claims from trusted issuers (provided default implementation)
+- **Merkle Tree**: Efficient verification using merkle proofs
+- **Zero-Knowledge**: Privacy-preserving verification with custom ZK circuits
+- **Other custom approaches**: Any implementation that satisfies the `verify_identity` interface
+
+The architecture of the default claim-based implementation is shown in the diagram below:
+
+```mermaid
+graph TB
+
+    RWA[RWA Token]
+    IDV[Identity Verifier]
+
+    subgraph " "
+        CTI[Claim Topics & Issuers]
+        IRS[Identity Registry Storage]
+        CI[Claim Issuer]
+        IC[Identity Claims]
+
+        IV_TITLE["Claim-Based Identity Verification"]
+        IV_TITLE ~~~ IRS
+    end
+
+
+
+    RWA -->|"verify_identity()<br/>recovery_target()"| IDV
+
+
+    IDV -->|"get_claim_topics_and_issuers()"| CTI
+    IDV -->|"stored_identity()<br/>get_recovered_to()"| IRS
+    IDV -->|"try_is_claim_valid()"| CI
+    IDV -->|"get_claim_ids_by_topic()<br/>get_claim()"| IC
+
+    CI --> |"has_claim_topic()"| CTI
+    IC --> |"is_claim_valid()"| CI
```

Comment:
`IdentityClaims` calls `is_claim_valid()` only when it adds a new claim which is a separate flow than the one pictured here.

Reply:
My goal was to include all the functions in here. There is no specific path is displayed here, but all possible connections instead